src/Impl/GuzzleHttpRequester.php

```diff
@@ -48,11 +48,13 @@ private function handleResponse(ResponseInterface $response): SuccessResult
         $statusCode = $response->getStatusCode();
 
         if ($statusCode == 202) {
-            $messageStatus = $this->parseMessageStatus($response);
-            if ($messageStatus->getSuccess() != null) {
-                return $messageStatus->getSuccess();
+            $messageStatuses = $this->parseMessageStatuses($response);
+            foreach ($messageStatuses as $messageStatus) {
+                if ($messageStatus->getFailure() != null) {
+                    throw $this->streamxClientExceptionFrom($messageStatus->getFailure()); // TODO: handle collecting all failures into this response exception
+                }
             }
-            throw $this->streamxClientExceptionFrom($messageStatus->getFailure());
+            return $messageStatuses[0]->getSuccess(); // TODO: handle returning list of statuses, not just the first success
```

Comment:
Some TODOs left, but it passes 100% tests (added sunny path integration test for the changes)

Reply:
It only means that we don't have the test that will fail. Processing just one status instead of all is something that has to be fixed.

Reply:
Yes, this PR in only a draft PR to receive the overview of places that need adjusting.

But it handles the sunny path when everything succeeds -> and serves (in this state) as a working PoC of sending a multi-message json to StreamX (in each publish and unpublish scenario). The TODOs are to be implemented. 

The goal is to use such multi-message batch publishing in the StreamX Magento Connector